pkg/sources/git/git.go

```diff
@@ -319,6 +321,43 @@ type cloneParams struct {
 	clonePath string
 }
 
+//Defines the interface for returning a single PID value given an executable name
+type PID interface {
+	getPIDs(ctx context.Context, executable string) (int, error)
+}
+
+var ExecutableName = "trufflehog"
+
+func GetScannerPIDs(ctx context.Context, executable string) (int, error)  {
+	var pids []int
+
+	procs, err := ps.Processes()
+	if err != nil {
+		return 0, fmt.Errorf("error getting jobs PIDs: %w", err)
+	}
+
+	for _, proc := range procs {
+		if strings.Contains(proc.Executable(), executable) {
+			pids = append(pids, proc.Pid())
+		}
+	}
+
+	if len(pids) < 2 {
+		return pids[0], nil
+	} else {
```

Comment:
**Suggestion:** This will panic if `len(pids) == 0`. Maybe something like this:

```go
if len(pids) == 0 {
    return 0, fmt.Errorf("PID for %q not found", executable)
} else if len(pids) == 1 {
    return pids[0], nil
} else {
    // ...
}
```